modules/component/src/componentNext.ts

```diff
@@ -483,6 +484,83 @@ export class ComponentNext<P1 extends ComponentProps> {
       }
     )
   }
+  toList<T extends string | number>(
+    fn: (p: iProps<P1>) => T
+  ): iComponentNext<
+    P1,
+    {
+      iState: ListComponentState<iState<P1>>
+      oState: ListComponentState<oState<P1>>
+      iActions: Action<iActions<P1>, T>
+      oActions: Action<oActions<P1>, T>
+    }
+  > {
+    return new ComponentNext(
+      () => ListComponentState.of(this._init as () => iState<P1>),
+      (inputAction: any, state: any) => {
```

Comment:
may be instead of using `any` you use — 

```ts
(inputAction: iAction<P1>, state: ListComponentState<iState<P1>>) => {
  /// ...
}
```

Reply:
We can't do that now because typings of `_update` is `(action: unknown, state: unknown) => unknown`